### PR TITLE
Use flex to align SVG and text

### PR DESCRIPTION
**Pull request by wxiaoguang**
The code can be as simple as:

```html
<div class="flex-text-block">{{svg "octicon-alert"}} {{svg "octicon-x"}} text (block)</div>
<div><div class="flex-text-inline">{{svg "octicon-alert"}} {{svg "octicon-x"}} text</div> (inline)</div>
<div><button class="ui red button">{{svg "octicon-alert" 24}} {{svg "octicon-x" 24}} text</button></div>
```

image

### CODE REVIEW

**Comment by silverwind** [comment]
Also fix [these cases](https://github.com/go-gitea/gitea/pull/25158#issuecomment-1583221095) here?

**Comment by wxiaoguang** [comment]
Yup, if this approach looks good to maintainers, I can do more improvements.

File: web_src/css/base.css [hunk 1]
```diff
@@ -2367,3 +2367,11 @@ table th[data-sortt-desc] .svg {
   width: 15px;
   height: 15px;
 }
+
+.ui.button.flex-text-wrap,
+.flex-text-wrap {
+  display: inline-flex;
```
Comment (silverwind) [inline web_src/css/base.css:2316]: `inline-flex` may introduce unexpected inline rendering, so I recommend `flex` (block mode). This is already one issue why you can't make such generic classes, sometimes one wants `inline`, sometimes one wants `block`.

Comment (wxiaoguang) [reply web_src/css/base.css:2316]: By default, most elements like "ui button" are "inline-block". So, "inline-flex" is good enough to replace "inline-block"

I have considered about this problem, IMO if we need to use "display: flex", we can do `flex-text-wrap gt-df`, then everything is still good.

Comment (wxiaoguang) [reply web_src/css/base.css:2316]: Or maybe we can use separate CSS names, eg: `flex-inline-wrap` vs `flex-block-wrap`

Comment (silverwind) [reply web_src/css/base.css:2316]: `flex-text-block` and `flex-text-inline` would be my preference. Most if not all text should wrap, so it doesn't need special attention in the name.

**Comment by wxiaoguang** [comment]
Now we can align everything easily, including dropdown icons:

<details>

image

</details>

If it looks good to you, I can continue.

File: web_src/css/base.css [hunk 2]
```diff
@@ -2328,3 +2295,57 @@ table th[data-sortt-desc] .svg {
...
+  line-height: 1em;
+}
+
+.ui.ui.dropdown > .icon.icon {
+  /* plain dropdown and button dropdown use flex layout for icons */
+  position: initial;
+  padding: 0;
+  margin: 0;
+  height: auto;
+}
```
Comment (silverwind) [inline web_src/css/base.css:2291]: `.icon.icon` is for specificity? Maybe comment on that.

**Comment by silverwind** [comment]
Seems good so far. These vertical centerings were in need of a rework anyways.

**Review by HesterG — APPROVED** [review]

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
This brought a regression on this button on repo homepage:

Screenshot 2023-06-14 at 21 43 31

**Comment by wxiaoguang** [comment]
I will take it. I have check that page but somehow I missed that .... 😭

**Comment by wxiaoguang** [comment]
->  Fix some UI alignments #25277

**Comment by silverwind** [comment]
Another regression I noticed:

image

@wxiaoguang any idea how to solve that one? I guess in worst case, we just remove the dropdown triangle.

**Comment by wxiaoguang** [comment]
It's caused by : `.repository .diff-detail-box .button { padding: 8px 12px; }`, but by default "ui dropdown" uses relative + absolute to position the dropdown icon.

In this case, I think the UI looks better without the dropdown icon.

If the dropdown icon should be kept, it could be something like "svg gitea-whitespace gt-mr-3"

See the screenshots in #25397

**Comment by silverwind** [comment]
Found another triangle regression on the commit page:

Screenshot 2023-06-22 at 14 19 37

**Comment by wxiaoguang** [comment]
->  Fine tune "dropdown button" icon #25442

**Pull request merged**